helios-services/src/main/java/com/spotify/helios/agent/AgentService.java

```diff
@@ -320,13 +326,39 @@ public AgentService(final AgentConfig config, final Environment environment)
    */
   private ZooKeeperClient setupZookeeperClient(final AgentConfig config, final String id,
                                                final CountDownLatch zkRegistrationSignal) {
+    ACLProvider aclProvider = null;
+    List<AuthInfo> authorization = null;
+
+    if (config.isZooKeeperEnableAcls()) {
+      final String agentPassword = config.getZooKeeperAgentPassword();
+      final String masterDigest = config.getZooKeeperMasterDigest();
+
+      if (agentPassword == null) {
+        throw new HeliosRuntimeException("ZooKeeper ACLs enabled but agent password not set");
+      }
+
+      if (masterDigest == null) {
+        throw new HeliosRuntimeException("ZooKeeper ACLs enabled but master digest not set");
+      }
+
+      final String agentDigest = Base64.toBase64String(Hash.sha1digest(
+          String.format("%s:%s", ZooKeeperAclProvider.AGENT_USER, agentPassword).getBytes()));
```

Comment:
Pass master-digest, agent-digest and the agent password? You could, but it's redundant since the agent-digest can be computed from the agent password.